README.md

```diff
@@ -4,222 +4,1008 @@
 
-## Examples
+* Uncategorised:
 
-There are documented examples within the [examples](https://github.com/SealabJaster/jcli/tree/master/examples) folder.
+    * All individual parts of this library are intended to be reusable. Allowing you to build your own CLI core using these already-made components, if desired.
 
-Other than there, here's a list of projects/scripts using JCLI:
+# Quick Start
 
-* [JCLI's test runner](https://github.com/SealabJaster/jcli/blob/master/examples/test.d)
+*This is a brief overview, for slightly more in-depth examples please look at the fully-documented [examples](https://github.com/BradleyChatha/jcli/tree/master/examples) folder.*
 
-* [JCLI's manual/misc test program](https://github.com/SealabJaster/jcli_testerr)
+## Creating a default command
 
-## Changelog
+The default command is the command that is ran when you don't specify the name of a named command. e.g. `mytool 60 20 --some=args` would call the default command if it exists:
 
-I publish changelogs over at [my website](https://bradley.chatha.dev/Blog/JcliNews).
+```d
+// inside of app.d
+module app;
+import jaster.cli;
 
-## Quick Start
+@Command(null, "The default command.")
+struct DefaultCommand
+{
+    int onExecute()
+    {
+        return 0;
+    }
+}
+```
 
-Include this library in your project (e.g. `dub add jcli`).
+The `@Command` is a UDA (User Defined Attribute) where the first parameter is the command's name (`null` for the default command), and the second parameter is the command's description.
 
-Create a class or struct; attach `@Command("commandname")` onto it; add as many public variables as you want, with `@CommandNamedArg` or `@CommandPositionalArg` attached; 
-add an `onExecute` function that returns either `void` or `int`:
+All commands must define an `onExecute` function, which either returns `void`, or an `int` that will be used as the program's exit/status code.
 
-```d
-module mytool.commands;
+As a side note, an initial dub project does not include the intial `module app;` shown in the example above. I've added it as we'll need the directly reference the module in a later section.
 
-import jaster.cli;
+## Positional Arguments
 
-// Matches: mytool compile all
-//          mytool compile
-//          mytool ct
-@Command("compile all|compile|ct")
-struct CompileCommand
-{
-    // Use `Nullable` for optional arguments.
-    // `Nullable` is publicly imported by JCLI for ease-of-use.
-    @CommandNamedArg("o|output", "Where to place the compiled output")
-    Nullable!string output;
+To start off, let's make our default command take a number as a positional arg. If this number is even then return `1`, otherwise return `0`.
 
-    @CommandPositionalArg(0, "The file to compile")
-    string toCompile;
+Positional arguments don't have a name, and are expected to exist in a specific position within the arguments passed to your program.
```

Comment:
My wording is definitely unclear here.

What I was trying to get at is that positional args don't need a name to function, and that they only exist for the user's benefit.

I might just cut the "don't have a name" part out though, since it's relatively unimportant (and incorrect).